src/scorer.py

```diff
@@ -55,6 +53,18 @@ def score(self, cand_o: Optional[str]) -> Tuple[int, str]:
         deletions = []
         insertions = []
 
+        def imm_matches_everything(row: str, arch: ArchSettings) -> bool:
+            if arch.name == "ppc":
+                return any(str in row for str in ["...data", "@"])
+            else:
+                return "(." in row
+
+        def is_valid_symbol(str: str) -> bool:
+            return (
+                re.sub(re.compile(r"[A-Za-z_$][A-Za-z0-9_$]*"), "", str.split("@")[0])
```

Comment:
there are functions in `re` for checking whether a regex matches another string, you don't have to do search-replace

Reply:
on it